include/static-pages.php

```diff
@@ -122,23 +124,25 @@ public function page_link( $link, $id ) {
 	}
 
 	/**
-	 * Adds page_on_front and page_for_posts properties to the language objects.
+	 * Adds page_on_front and page_for_posts properties to language data before the object is created.
 	 *
-	 * @since 1.8
+	 * @since 3.4
 	 *
-	 * @param PLL_Language[] $languages The list of languages.
-	 * @param PLL_Model      $model     The instance of PLL_Model.
-	 * @return PLL_Language[]
+	 * @param array $static_pages Array of language page_on_front and page_for_posts properties.
+	 * @param array $language Language data.
+	 * @return array Language data with static pages ids.
+	 *
+	 * @phpstan-return array{page_on_front: int<0, max>, page_for_posts: int<0, max>}
 	 */
-	public static function pll_languages_list( $languages, $model ) {
+	public function set_static_pages( $static_pages, $language ) {
 		if ( 'page' === get_option( 'show_on_front' ) ) {
-			foreach ( $languages as $k => $language ) {
-				$languages[ $k ]->page_on_front = $model->post->get( get_option( 'page_on_front' ), $language );
-				$languages[ $k ]->page_for_posts = $model->post->get( get_option( 'page_for_posts' ), $language );
-			}
+				$page_on_front_translations     = $this->model->post->get_translations_from_term( get_option( 'page_on_front' ) );
```

Comment:
Since `get_option()` can return `false` (or `mixed` according to the doc), I would check its result before calling `get_translations_from_term()`. This way PHPStan should be happy.

Reply:
Done in 1d5dcf3f3fb67570577444c331a3416724e0e433